src/main/java/com/cs544/project/domain/Student.java

```diff
@@ -25,7 +23,10 @@ public class Student extends Person {
     @Column(name = "ApplicantID")
     private String applicantID;
 
-    @ManyToOne
+    @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "FacultyAdviserID")
     private Faculty faculty;
+
+    @OneToMany(mappedBy = "student")
+    private List<CourseRegistration> courseRegistrationList;
```

Comment:
i don't think we should have a composition of List<CourseRegistration>. It's not part of the DR diagram and it's not performant (since if I want a student only, I have to get their registrations too, even if I don't want them — can lazy load, yes. but just adds complexity we can simplify) 

to get the student's courses, it is better (imo) to have a "getCourseRegistrationByStudentId()" method in the repository so it can provide the list of registrations since it already owns that relationship

@bishosilwal

Reply:
yes, we can change it according to our need.